src/main/java/seedu/address/model/person/Person.java

```diff
@@ -80,6 +82,28 @@ public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
+    /**
+     * Creates a new Person object with the given parameters.
+     * @param name The name of the person.
+     * @param phone The phone number of the person.
+     * @param email The email of the person.
+     * @param role The role of the person.
+     * @param address The address of the person.
+     * @param course The course of the person.
+     * @param tags The tags of the person.
+     * @return A new Person object.
+     * @throws CommandExecutionException If the address is invalid.
+     */
+    public static Person createPerson(Name name, Optional<Phone> phone, Email email, Role role,
+                                      Optional<Address> address, Course course, Set<Tag> tags)
+            throws CommandExecutionException {
+        try {
+            return new Person(name, phone, email, role, address, course, tags);
+        } catch (InvalidAddressException e) {
+            throw new CommandExecutionException(Address.MESSAGE_CONSTRAINTS_INVALID_PROFESSOR);
```

Comment:
`CommandExecutionException` is for **commands**, it doesn't make sense to catch and re-throw it here in `model` directory.

Plus, why is the message like that...?

Reply:
OK I'll make it make more sense. Can you be more clear what you mean by "like that"?